internal/querycoordv2/job/job.go

```diff
@@ -0,0 +1,487 @@
+	broker meta.Broker,
+	nodeMgr *session.NodeManager,
+) *LoadCollectionJob {
+	return &LoadCollectionJob{
+		BaseJob:   NewBaseJob(ctx, req.Base.GetMsgID(), req.GetCollectionID()),
+		req:       req,
+		dist:      dist,
+		meta:      meta,
+		targetMgr: targetMgr,
+		broker:    broker,
+		nodeMgr:   nodeMgr,
+	}
+}
+
+func (job *LoadCollectionJob) PreExecute() error {
+	req := job.req
+	log := log.With(
+		zap.Int64("msgID", req.Base.GetMsgID()),
+		zap.Int64("collection", req.GetCollectionID()),
+	)
+
+	if req.GetReplicaNumber() <= 0 {
+		log.Info("request doesn't indicate the number of replicas, set it to 1",
+			zap.Int32("replica-number", req.GetReplicaNumber()))
+		req.ReplicaNumber = 1
+	}
+
+	if job.meta.Exist(req.GetCollectionID()) {
+		old := job.meta.GetCollection(req.GetCollectionID())
+		if old == nil {
+			msg := "collection with different load type existed, please release it first"
+			log.Warn(msg)
+			return utils.WrapError(msg, ErrLoadParameterMismatched)
+		}
+		if old.GetReplicaNumber() != req.GetReplicaNumber() {
+			msg := "collection with different replica number existed, release this collection first before changing its replica number"
+			log.Warn(msg)
+			return utils.WrapError(msg, ErrLoadParameterMismatched)
+		}
+		return ErrCollectionLoaded
+	}
+
+	return nil
+}
+
+func (job *LoadCollectionJob) Execute() error {
+	req := job.req
+	log := log.With(
+		zap.Int64("msgID", req.GetBase().GetMsgID()),
+		zap.Int64("collection", req.GetCollectionID()),
+	)
+
+	// Create replicas
+	// TODO(yah01): store replicas and collection atomically
+	err := utils.SpawnReplicas(job.meta.ReplicaManager,
+		job.nodeMgr,
+		req.GetCollectionID(),
+		req.GetReplicaNumber())
+	if err != nil {
+		msg := "failed to spawn replica for collection"
+		log.Error(msg, zap.Error(err))
+		return utils.WrapError(msg, err)
+	}
+
+	// Fetch channels and segments from DataCoord
+	partitions, err := job.broker.GetPartitions(job.ctx, req.GetCollectionID())
+	if err != nil {
+		msg := "failed to get partitions from RootCoord"
+		log.Error(msg, zap.Error(err))
+		return utils.WrapError(msg, err)
+	}
+
+	collection := &meta.Collection{
+		CollectionLoadInfo: &querypb.CollectionLoadInfo{
+			CollectionID:  req.GetCollectionID(),
+			ReplicaNumber: req.GetReplicaNumber(),
+			Status:        querypb.LoadStatus_Invalid,
+		},
+		CreatedAt: time.Now(),
+	}
+
+	// Make sure the handoff event before this fetches segments can be handled
+	job.meta.CollectionManager.PutCollectionWithoutSave(collection)
+	err = utils.RegisterTargets(job.ctx,
+		job.targetMgr,
+		job.broker,
+		req.GetCollectionID(),
+		partitions...)
+	if err != nil {
+		msg := "failed to register channels and segments"
+		log.Error(msg, zap.Error(err))
+		return utils.WrapError(msg, err)
+	}
+
+	collection = collection.Clone()
+	collection.Status = querypb.LoadStatus_Loading
+	err = job.meta.CollectionManager.PutCollection(collection)
+	if err != nil {
+		msg := "failed to store collection"
```

Comment:
log will be valueless if lacks key information.

Reply:
logger is initialed at the begining.